### ISSUE TITLE
Accessibility Help: Don't use monaco-editor, but semantic html

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#187587]
The accessibility help menu uses a monaco-editor to present the help text:

![Code_-_Insiders_ZZmWt6dE4v](https://github.com/microsoft/vscode/assets/2931520/ff5416dd-646f-44b2-8e7d-5609f2254fd6)

I think we should try to use plain (semantic) html instead:

* I'm not so sure that the text editor actually improves accessibility (but I might be wrong).
   * The web offers many accessibility features and is optimized for rendering documents to read. I'd be very suprised if putting a document in a monaco-editor would give better accessibility features than using native web-features (e.g. `<p>`, `<h1>` elements)
   * NVDA has a special browse mode to navigate between headings (using `h`) and sentences (using arrow keys). If this mode does not work in VS Code, we should try to fix it.
   * I don't think character-level navigation is that useful for reading paragraphs.
   * [The NVDA help itself is a static html document that is not in a text editor](https://www.nvaccess.org/files/nvda/documentation/userGuide.html)
* It is inconsistent with other places where we render text (e.g. the trusted workspaces dialog)

FYI @isidorn

**Comment by isidorn** [comment]
Thanks @hediet for opening this issue.
We should figure out what is the main benefit of using the Monaco editor compared to a native HTML element.

fyi @jooyoungseo @jvesouza

**Comment by meganrogge** [comment]
The main benefit is users do not have to switch between modes to inspect content vs navigate. Inspection on a char by char, word by word, sentence by sentence basis is very beneficial to screen reader users. That can be done here using only arrow keys rather than switching modes and recalling the exact keystrokes required to do so for html elements.

**Comment by rperez030** [comment]
> The main benefit is users do not have to switch between modes to inspect content vs navigate. Inspection on a char by char, word by word, sentence by sentence basis is very beneficial to screen reader users. That can be done here using only arrow keys rather than switching modes and recalling the exact keystrokes required to do so for html elements.

If the semantic HTML is wrapped in a container with role ="document", NVDA and JAWS will be smart enough to switch back and forth between virtual mode and application mode, and the keystrokes required to interact with web content are used all the time. That is not the problem in my opinion, although I'm aware there are different views about this.

The question here is perhaps how much value there is in exposing rich semantic content in the context of accessibility help. are there links, tables or a complex heading structure that makes this level of navigation helpful? What about sighted keyboard users? Is it necessary to have a cursor in this context to select text, for example? What is the screen reader user, or anyone else, missing right now by presenting the information as text only?

If we enable browse mode in this context, we may need to implement it like a modal dialog, for example, to prevent the browse cursor on windows from navigating beyond the boundaries of the contextual help. In addition, Voiceover in macOS doesn't have a browse mode equivalent. The navigation paradigm in that platform is different.

In the context of macOS, the current implementation is very convenient because there is a cursor that the screen reader user can use to explore the document. If we present that as web content without a caret, we would be forcing voiceover users to use a different approach required to explore that content which is usually more complex. Is it worth it?

**Comment by meganrogge** [comment]
Thanks @rperez030 for your insight

@hediet and I discussed his concerns which do make sense - we want to make sure that we're not adding so much content without a convenient way to navigate it. 

We can keep this open for future discussion

**Comment by jooyoungseo** [comment]
@meganrogge IMO, "document" role markup might work for this case. I know, we are reluctant to go with "document" role because of the unsatisfactory result that we experienced in the terminal buffer area, which had complicated and nested dynamic content. I think help menu is much simpler and worth trying with "document" landmark. Another workaround is allowing Alt+F2 to work in the help menu for some screen readers that do not support document role very well.

**Comment by meganrogge** [comment]
Thanks for your insight @jooyoungseo. Since I just reworked these menus to use the Monaco editor instance, and there aren't accessibility issues or perf drawbacks of this approach, I will probably wait awhile before making more changes to this.

**Comment by jooyoungseo** [comment]
@meganrogge -- Great! My personal preference is still a Monaco, but just in case the help menu contains any hyper links. Aw, actually, this discussion has to do with the terminal buffer: How can we augment semantic/HTML content in Monaco in the long run? Just dirty and funny idea: Could we have a similar feature like Alt+F2 that displays currently focused content in "document" landmark webview?

**Comment by meganrogge** [comment]
@jooyoungseo there is an editor command via monaco `Open Link` which provides this support. It opens the link closest to one's cursor ATM. We could tweak it to provide all links if that would be more helpful

**Comment by meganrogge** [comment]
@amunger is working on using the accessible view for notebook output, which can be quite large. 

I offered one idea - which is we could register a language in such cases that would have symbol providers similar to what we have in the editor. That way, screen reader users wouldn't have to change modes, could still use arrow keys for navigation, and yet could jump to important parts.

**Comment by hediet** [comment]
Because of the line-wraps, I can imagine that the line-by-line reading can be quite confusing in some cases:

![Code_-_Insiders_hC62mxcvsh](https://github.com/microsoft/vscode/assets/2931520/667c74f3-be86-4136-8fb0-8a3c358b5b13)

Often, the first line can stand by itself (*The code above is a TypeScript implementation of a method ... that takes an array*) and then the next line (*of `ILineChange` objects*) is confusing (the screenreader voice uses intonation to indicate the end of a sentence, which causes even more confusion here).

I think it would make more sense to chunk the text into logical segments (like sentences, relative-clauses, etc.) and allow for navigation between them. This would be much easier in an html-based implementation than in the monaco-editor.

**Comment by meganrogge** [comment]
We now have go to symbol which allows for navigating between HTML elements, so I believe this issue can be closed.
